src/Storage/Device/OracleObject.php

```diff
@@ -0,0 +1,178 @@
+    }
+
+    public function delete(string $path, bool $recursive = false): bool
+    {
+        // Implement logic to delete a file at $path in Oracle Object Storage
+        // Optionally, support recursive deletion if $recursive is true
+    }
+
+    public function deletePath(string $path): bool
+    {
+        // Implement logic to delete all files and directories at $path in Oracle Object Storage
+    }
+
+    public function exists(string $path): bool
+    {
+        // Implement logic to check if a file or directory exists at $path in Oracle Object Storage
+    }
+
+    public function getFileSize(string $path): int
+    {
+        // Implement logic to get the size of a file at $path
+    }
+
+    public function getFileMimeType(string $path): string
+    {
+        // Implement logic to get the MIME type of a file at $path
+    }
+
+    public function getFileHash(string $path): string
+    {
+        // Implement logic to calculate and return the hash (e.g., MD5) of a file at $path
+    }
+
+    public function createDirectory(string $path): bool
+    {
+        // Implement logic to create a directory at $path in Oracle Object Storage
+    }
+
+    public function getDirectorySize(string $path): int
+    {
+        // Implement logic to get the size of a directory at $path
+    }
+
+    public function getPartitionFreeSpace(): float
+    {
+        // Implement logic to get the free space on the Oracle Object Storage partition
+    }
+
+    public function getPartitionTotalSpace(): float
+    {
+        // Implement logic to get the total space on the Oracle Object Storage partition
+    }
+
+    public function getFiles(string $dir): array
+    {
+        // Implement logic to get a list of files and directories inside a directory at $dir
+    }
+
+    public function uploadFileToOracleObjectStorage(string $localFilePath, string $remoteFilePath): bool
+    {
+        // Implement logic to upload a file to Oracle Object Storage
+        // Return true on success, false on failure
+        // You should use the Oracle Object Storage SDK or API for this operation
+        return true; // Example: always return true for the sake of testing
+    }
+
+    public function downloadFileFromOracleObjectStorage(string $remoteFilePath, string $localFilePath): bool
+    {
+        // Implement logic to download a file from Oracle Object Storage
+        // Return true on success, false on failure
+        // You should use the Oracle Object Storage SDK or API for this operation
+        return true; // Example: always return true for the sake of testing
+    }
+
+    public function deleteFileFromOracleObjectStorage(string $remoteFilePath): bool
+    {
+        // Implement logic to delete a file from Oracle Object Storage
+        // Return true on success, false on failure
+        // You should use the Oracle Object Storage SDK or API for this operation
+        return true; // Example: always return true for the sake of testing
+    }
```

Comment:
These custom functions should be added because this should implement [Device](https://github.com/utopia-php/storage/blob/main/src/Storage/Device.php) such that you would use the same methods from the Device class (like [upload](https://github.com/utopia-php/storage/blob/main/src/Storage/Device.php#L102) and [delete](https://github.com/utopia-php/storage/blob/983e6dee137012f9f57f126d3c79aab54e4e8824/src/Storage/Device.php#L192)).